src/backend/base/langflow/components/vectorstores/OpenSearch.py

```diff
@@ -0,0 +1,292 @@
+                documents = []
+                for data in self.ingest_data:
+                    if isinstance(data, Data):
+                        doc = data.to_lc_document()
+                        documents.append(doc)
+                    else:
+                        error_message = f"Expected Data object, got {type(data)}"
+                        logger.error(error_message)
+                        raise ValueError(error_message)
+
+                # Create OpenSearchVectorSearch instance using from_documents method
+                try:
+                    opensearch = OpenSearchVectorSearch.from_documents(
+                        documents,
+                        self.embedding,
+                        opensearch_url=self.opensearch_url,
+                        http_auth=(self.username, self.password),
+                        use_ssl=self.use_ssl,
+                        verify_certs=self.verify_certs,
+                        ssl_assert_hostname=False,
+                        ssl_show_warn=False,
+                        index_name=self.index_name,
+                    )
+                    return opensearch
+                except Exception as e:
+                    logger.error(f"Failed to create OpenSearchVectorSearch instance: {str(e)}")
+                    raise RuntimeError(
+                        "Error creating OpenSearchVectorSearch instance. Check your connection and credentials."
+                    ) from e
+            else:
+                # No ingest_data provided, connect to existing index
+                try:
+                    opensearch = OpenSearchVectorSearch(
+                        index_name=self.index_name,
+                        embedding_function=self.embedding,
+                        opensearch_url=self.opensearch_url,
+                        http_auth=(self.username, self.password),
+                        use_ssl=self.use_ssl,
+                        verify_certs=self.verify_certs,
+                        ssl_assert_hostname=False,
+                        ssl_show_warn=False,
+                    )
+                    return opensearch
+                except Exception as e:
+                    logger.error(f"Failed to connect to OpenSearchVectorSearch instance: {str(e)}")
+                    raise RuntimeError(
+                        "Error connecting to OpenSearchVectorSearch instance. Check your connection and credentials."
+                    ) from e
+        except Exception as e:
+            logger.error(
+                f"Unexpected error in build_vector_store, checkout if the username and the password were correctly set: {str(e)}"
+            )
+            raise RuntimeError(
+                "An unexpected error occurred while building the vector store, checkout if the username and the password were correctly set."
+            ) from e
+
+    def _add_documents_to_vector_store(self, vector_store: "OpenSearchVectorSearch") -> None:
```

Comment:
I don't actually see this being used anywhere. I see that it's probably based off of the other vector stores impls, but since you already have a separate path for creating the vector store `from_documents`, you didn't have a need for this. Would either remove this, or update the `build_vector_store` method to create the store without documents, then use this method to add them after creation .

Reply:
Yes, I fixed this and now I am using it to add the data when ingest data is passed to the component.